File: src/main/java/jaba/set/Set.java
package jaba.set;

public interface Set<Item> {
    /**
     * Tells if the set contains the specified item.
     */
    boolean contains(Item item);

    /**
     * Provides the number of elements in the set.
     */
    int size();

    /**
     * Tells if the set is empty.
     */
    boolean isEmpty();
}

File: src/main/java/typescript/set/set.ts
export interface Set<Item> {
    /**
     * Tells if the set contains the specified item.
     */
    contains(item:Item):boolean;

    /**
     * Provides the number of elements in the set.
     */
    size():number;

    /**
     * Tells if the set is empty.
     */
    isEmpty():boolean;
}